### PR TITLE
Naming some functions in libultra_code, decompiling a decent amount.

### PR DESCRIPTION
**Pull request by shawlucas**
Decompiled sinf.c, sins.c, cosf.c, coss.c sp.c, sptask.c, __osSpRawStartDma.c, some others ?

### CODE REVIEW

File: src/libultra_code/cosf.c [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+ * ====================================================================
+ */
+
+float fcos(float x) {
+    float absx;
+    double dx, xsq, poly;
+    double dn;
+    int n;
+    double result;
+    int ix, xpt;
```
Comment (Random06457) [inline src/libultra_code/cosf.c:38]: types from `types.h` should be used here and below
```suggestion
f32 fcos(f32 x) {
    f32 absx;
    f64 dx, xsq, poly;
    f64 dn;
    s32 n;
    f64 result;
    s32 ix, xpt;
```

File: src/libultra_code/coss.c [hunk 2]
```diff
@@ -0,0 +1,6 @@
+#include "libultra_internal.h"
+
+signed short coss(unsigned short x) {
+    /* 0 <= x < 0x10000  ==>  0 <= x < 2PI */
+    return sins((unsigned short)(x + 0x4000));
```
Comment (Random06457) [inline src/libultra_code/coss.c:5]:
```suggestion
s16 coss(u16 x) {
    /* 0 <= x < 0x10000  ==>  0 <= x < 2PI */
    return sins((u16)(x + 0x4000));
```

File: src/libultra_code/sinf.c [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+ *
+ * ====================================================================
+ */
+
+float fsin(float x) {
+    double dx, xsq, poly;
+    double dn;
+    int n;
+    double result;
+    int ix, xpt;
```
Comment (Random06457) [inline src/libultra_code/sinf.c:37]:
```suggestion
f32 fsin(f32 x) {
    f64 dx, xsq, poly;
    f64 dn;
    s32 n;
    f64 result;
    s32 ix, xpt;
```

File: src/libultra_code/sins.c [hunk 4]
```diff
@@ -0,0 +1,22 @@
+#include "guint.h"
+#include "sintable.h"
+
+signed short sins(unsigned short x) {
+    /* 0 <= x < 0x10000  ==>  0 <= x < 2PI */
+
+    signed short val;
```
Comment (Random06457) [inline src/libultra_code/sins.c:7]:
```suggestion
s16 sins(u16 x) {
    /* 0 <= x < 0x10000  ==>  0 <= x < 2PI */

    s16 val;
```

File: asm/non_matchings/code/code_801031F0/func_801032B0.s [hunk 5]
```diff
@@ -1,69 +1,12 @@
-.include "macro.inc"
+.rdata
```
Comment (Roman971) [inline asm/non_matchings/code/code_801031F0/func_801032B0.s:1]: .rdata is specifically for strings, while .late_rodata is for the rest of rodata like floats/doubles.
```suggestion
.late_rodata
```

File: asm/non_matchings/code/cosf/cosf.s [hunk 6]
```diff
@@ -0,0 +1,117 @@
+.rdata
```
Comment (Roman971) [inline asm/non_matchings/code/cosf/cosf.s:1]: same here about .late_rodata vs .rdata
```suggestion
.late_rodata
```

File: asm/non_matchings/code/sins/sins.s [hunk 7]
```diff
@@ -0,0 +1,41 @@
+.data
```
Comment (Roman971) [inline asm/non_matchings/code/sins/sins.s:1]: .data doesn't belong in non matching files, it's better to keep it in a separate .data.s file in data/ until it's decompiled.

File: include/ultra64/controller.h [hunk 8]
```diff
@@ -6,6 +6,38 @@
...
+#define CONT_CMD_READ_BUTTON_RX 4
+#define CONT_CMD_READ_MEMPACK_RX 33
+#define CONT_CMD_WRITE_MEMPACK_RX 1
+#define CONT_CMD_READ_EEPROM_RX 8
+#define CONT_CMD_WRITE_EEPROM_RX 1
+#define CONT_CMD_RESET_RX 3
+
+#define CONT_CMD_NOP 0xff
+#define CONT_CMD_END 0xfe //indicates end of a command
+#define CONT_CMD_EXE 1    //set pif ram status byte to this to do a command
```
Comment (Roman971) [inline include/ultra64/controller.h:35]:
```suggestion
#define CONT_CMD_NOP 0xFF
#define CONT_CMD_END 0xFE // indicates end of a command
#define CONT_CMD_EXE 1    // set pif ram status byte to this to do a command
```

File: include/ultra64/controller.h [hunk 9]
```diff
@@ -6,6 +6,38 @@
 #include <PR/os_cont.h>
 #include <PR/os_message.h>
 
+#define CONT_CMD_REQUEST_STATUS 0
+#define CONT_CMD_READ_BUTTON 1
+#define CONT_CMD_READ_MEMPACK 2
+#define CONT_CMD_WRITE_MEMPACK 3
+#define CONT_CMD_READ_EEPROM 4
+#define CONT_CMD_WRITE_EEPROM 5
+#define CONT_CMD_RESET 0xff
```
Comment (Roman971) [inline include/ultra64/controller.h:15]:
```suggestion
#define CONT_CMD_RESET 0xFF
```

File: src/libultra_code/sptask.c [hunk 10]
```diff
@@ -0,0 +1,66 @@
+#include "libultra_internal.h"
+#include <ultra64/hardware.h>
+
+#define _osVirtualToPhysical(ptr)              \
+    if (ptr != NULL) {                         \
+        ptr = (void*)osVirtualToPhysical(ptr); \
+    }
+
+static OSTask tmp_task;
```
Comment (Roman971) [inline src/libultra_code/sptask.c:9]: This name doesn't fit our naming scheme so it needs some adjustments.
```suggestion
static OSTask sTmpTask;

```

File: src/libultra_code/libultra_internal.h [hunk 11]
```diff
@@ -43,4 +43,6 @@ s32 __osAiDeviceBusy();
 void __osDispatchThread();
 u32 __osGetCause();
 s32 __osAtomicDec(u32*);
+signed short sins(unsigned short x);
+signed short coss(unsigned short x);
```
Comment (Roman971) [inline src/libultra_code/libultra_internal.h:47]: I don't think these prototypes belong here, but either way they should use the proper ultra types.
```suggestion
s16 sins(u16 x);
s16 coss(u16 x);
```

File: spec [hunk 12]
```diff
@@ -508,17 +508,15 @@ beginseg
     include "build/data/code_800FFA50.bss.o"
     include "build/asm/code_80100040.o"
     include "build/asm/code_801001E0.o"
-    include "build/asm/code_80100290.o"
-    include "build/data/code_80100290.data.o"
-    include "build/data/code_80100290.rodata.o"
-    include "build/data/code_80100290.bss.o"
```
Comment (Roman971) [inline spec:514]: I may just be missing them but I don't see those files removed in the PR, which they should be if they are no longer used in the spec.

**Comment by shawlucas** [comment]
Will fix when I get the chance later today

**Comment by shawlucas** [comment]
Should be fixed

**Pull request merged**